lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -98,6 +99,29 @@ static FloatAttr getF64FloatAttr(MLIRContext *context, double value) {
   return FloatAttr::get(Float64Type::get(context), value);
 }
 
+static llvm::Optional<Value> getScalarValue(Operation *base_op, Location loc,
+                                            PatternRewriter &rewriter) {
+  llvm::Optional<Value> scalar;
+  if (ValueTensorLiteralOp valueTensorLiteralOp =
+          dyn_cast<ValueTensorLiteralOp>(base_op)) {
+    if (valueTensorLiteralOp &&
+        getTensorRank(valueTensorLiteralOp.getResult()) == 0) {
+      auto val = valueTensorLiteralOp.value()
+                     .cast<ElementsAttr>()
+                     .cast<DenseElementsAttr>()
+                     .getSplatValue<int64_t>();
```

Comment:
The tensor might not be of type `int`. You should check the type before getting the splat value

Reply:
It would be great, If I get any hint for this. I tried multiple things but not a success.